controller/gke-cluster-config-handler.go

```diff
@@ -0,0 +1,349 @@
+	}
+
+	config = config.DeepCopy()
+	config.Status.Phase = gkeConfigCreatingPhase
+	return h.gkeCC.UpdateStatus(config)
+}
+
+func (h *Handler) checkAndUpdate(config *gkev1.GKEClusterConfig) (*gkev1.GKEClusterConfig, error) {
+
+	//TODO: implement -- currently a noop.
+	return config, nil
+}
+
+func (h *Handler) waitForCreationComplete(config *gkev1.GKEClusterConfig) (*gkev1.GKEClusterConfig, error) {
+	lastMsg := ""
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for {
+		svc, err := getServiceClient(ctx, config.Spec.CredentialContent)
+		if err != nil {
+			return config, err
+		}
+		cluster, err := svc.Projects.Locations.Clusters.Get(clusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName)).Context(ctx).Do()
+		if err != nil {
+			return config, err
+		}
+		if cluster.Status == runningStatus {
+			log.Infof(ctx, "Cluster %v is running", config.Spec.ClusterName)
+			config = config.DeepCopy()
+			config.Status.Phase = gkeConfigActivePhase
+			return h.gkeCC.UpdateStatus(config)
+		}
+		if cluster.Status != lastMsg {
+			log.Infof(ctx, "%v cluster %v......", strings.ToLower(cluster.Status), config.Spec.ClusterName)
+			lastMsg = cluster.Status
+		}
+		time.Sleep(time.Second * 5)
+	}
+
+}
+
+func waitClusterRemoveExp(ctx context.Context, svc *gkeapi.Service, config *gkev1.GKEClusterConfig) (*gkeapi.Operation, error) {
+	var operation *gkeapi.Operation
+	var err error
+
+	for i := 1; i < 12; i++ {
+		time.Sleep(time.Duration(i*i) * time.Second)
+		operation, err = svc.Projects.Locations.Clusters.Delete(clusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName)).Context(ctx).Do()
+		if err == nil {
+			return operation, nil
+		} else if !strings.Contains(err.Error(), "Please wait and try again once it is done") {
+			break
+		}
+	}
+	return operation, err
+}
+
+func getClientset(cluster *gkeapi.Cluster, ts oauth2.TokenSource) (kubernetes.Interface, error) {
+	capem, err := base64.StdEncoding.DecodeString(cluster.MasterAuth.ClusterCaCertificate)
+	if err != nil {
+		return nil, err
+	}
+	host := cluster.Endpoint
+	if !strings.HasPrefix(host, "https://") {
+		host = fmt.Sprintf("https://%s", host)
+	}
+
+	config := &rest.Config{
+		Host: host,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: capem,
+		},
+		WrapTransport: func(rt http.RoundTripper) http.RoundTripper {
+			return &oauth2.Transport{
+				Source: ts,
+				Base:   rt,
+			}
+		},
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func generateClusterCreateRequest(config *gkev1.GKEClusterConfig) *gkeapi.CreateClusterRequest {
+	request := &gkeapi.CreateClusterRequest{
+		Cluster: &gkeapi.Cluster{
+			NodePools: []*gkeapi.NodePool{},
+		},
+	}
+	request.Cluster.Name = config.Spec.ClusterName
+	request.Cluster.InitialClusterVersion = *config.Spec.KubernetesVersion
```

Comment:
At this point it is possible this is nil, you may want to add validation for create input.